server/src/main/java/org/spine3/server/entity/Repository.java

```diff
@@ -294,4 +307,44 @@ public int getIndex() {
             return result;
         }
     }
+
+    /**
+     * An iterator of all entities from the storage.
+     */
+    private static class EntityIterator<I, E extends Entity<I, ?>>
```

Comment:
Here we rely on having an index of identifiers from underlying storage. It can be implemented for any storage, not only in-memory.